src/detect-engine-loader.c

```diff
@@ -577,8 +577,6 @@ static TmEcode DetectLoaderThreadInit(ThreadVars *t, const void *initdata, void
     *data = ftd;
 
     DetectLoaderControl *loader = &loaders[ftd->instance];
-    SCMutexLock(&loader->m);
```

Comment:
yes this can be removed, not sure how I missed it

Reply:
I mentioned this in your PR BUT it was AFTER I approved it 🤦🏽‍♀️ so it went unnoticed, I think